src/main/java/org/openlca/ilcd/flows/epd/matml/PropertyDetails.java

```diff
@@ -0,0 +1,107 @@
+package org.openlca.ilcd.flows.epd.matml;
+
+import jakarta.xml.bind.annotation.XmlAccessType;
+import jakarta.xml.bind.annotation.XmlAccessorType;
+import jakarta.xml.bind.annotation.XmlAttribute;
+import jakarta.xml.bind.annotation.XmlElement;
+import jakarta.xml.bind.annotation.XmlElements;
+import jakarta.xml.bind.annotation.XmlID;
+import jakarta.xml.bind.annotation.XmlType;
+import org.openlca.ilcd.Vocab;
+import org.openlca.ilcd.commons.Copyable;
+import org.openlca.ilcd.util.Val;
+
+import java.util.ArrayList;
+import java.util.Optional;
+
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlType(name = "PropertyDetails	", propOrder = {"name", "units"})
+public class PropertyDetails implements Copyable<PropertyDetails> {
+
+	@XmlElement(name = "Name", namespace = Vocab.MATML, required = true)
+	private Name name;
+
+	@XmlElements(value = {
+		@XmlElement(
+			name = "Units",
+			namespace = Vocab.MATML,
+			type = Units.class,
+			required = true
+		),
+		@XmlElement(
+			name = "Unitless",
+			namespace = Vocab.MATML,
+			type = Unitless.class,
+			required = true
+		),
+	})
+	private Object units;
+
+	@XmlID
+	@XmlAttribute(name = "id", required = true)
+	private String id;
+
+	// region getters
+
+	public Name getName() {
+		return name;
+	}
+
+	public Object getUnits() {
+		return units;
+	}
+
+	public String getId() {
+		return id;
+	}
+
+	// endregion
+
+	// region setters
+
+	public PropertyDetails withName(Name name) {
+		this.name = name;
+		return this;
+	}
+
+	public PropertyDetails withUnits(Object units) {
+		this.units = units;
+		return this;
+	}
+
+	public PropertyDetails withId(String id) {
+		this.id = id;
+		return this;
+	}
+
+	public Name withName() {
+		if (name == null) {
+			name = new Name();
+		}
+		return name;
+	}
+
+	/**
+	 * Lazily set {@code PropertyDetails.units} as a new {@link Units} if not set
+	 * already.
+	 * Otherwise, returns {@code units} of type {@link Units} or {@link Unitless}.
+	 */
+	public Object withUnits() {
+		if (units == null) {
+			units = new Units();
+		}
+		return units;
+	}
```

Comment:
I was not sure how to default the `units` field here as it can be set to a `Unitless` instance.